docs/development.md

```diff
@@ -0,0 +1,31 @@
+# Developing on the-build
+
+To develop and test changes to the-build, you'll generally need to have a Drupal codebase to run the-build on.
+
+1. Choose a site to test against
+2. Check out the-build as a git repository into the site's `vendor/` directory
+3. Make, commit, and push your changes from within `vendor/palantirnet/the-build/`
+
+The major thing to watch out for here is that your copy of the-build is temporary -- in certain cases (especially when switching branches) when you run composer commands, your repo may be replaced with a different version.
+
+## Testing on an existing site
+
+You can clone the-build into the vendor directory of an existing site. If it doesn't have a version of the-build already installed, you'll need to require it with `composer require palantirnet/the-build:dev-release-2.0 --prefer-source --dev` first.
+
+```
+cd my-site
+composer require palantirnet/the-build:dev-release-2.0 --prefer-source --dev
+```
+
+At this point, `vendor/palantirnet/the-build` should be a clone of `git@github.com:palantirnet/the-build.git`.
+
+## Creating a new site for testing
+
+Follow the [Quick Start guide for palantirnet/drupal-skeleton](https://github.com/palantirnet/drupal-skeleton#quick-start). Before installing the-build in step 3, update the required version of the-build:
```

Comment:
I ran into some composer dependency version collisions, namely: `the-build` requiring `drupal/coder:~8.2.12` and `drupal-skeleton` requiring `drupal/coder:^8.0`

Here are links to the `composer.json` package definition for each: 
* `the-build`: https://github.com/palantirnet/the-build/blob/release-1.0/composer.json#L16
* `drupal-skeleton`: https://github.com/palantirnet/drupal-skeleton/blob/drupal8/composer.json#L41

Since `the-build` is a dependency requirement of `drupal-skeleton` and requires `drupal/coder`, I'm curious if it's necessary to have `drupal/coder` also listed as a dev dependency in `drupal-skeleton`?

Once I manually did `composer require drupal/coder:~8.2.12` in my new site I was able to insntall the rest of the dependencies, including requiring the `release-2.0` branch from `the-build`

I did have some other issues continuing with the `drupal-skeleton` quick start steps, but I'll leave notes over in that repo.